Incompatibility between `setter(strip_option)` and `setter(each)`

Combining  `setter(strip_option` and `setter(each)` to provide an "append" API for an _optional_ field fails to compile:

```
error[E0599]: no method named `extend` found for mutable reference `&mut Option<Vec<i32>>` in the current scope
```
### Example 

Provide a builder method `FooBuilder.bar` for appending to `FooBuilder.bars` one item at a time whilst defaulting to `None` if never called.

```rust
#[macro_use]
extern crate derive_builder;

#[derive(Debug, PartialEq, Default, Builder)]
struct Foo {
    #[builder(default, setter(strip_option, each = "bar"))]
    bars: Option<Vec<i32>>,
}

fn main() {
    assert_eq!(
        FooBuilder::default().build().unwrap(),
        Foo { bars: None },
        "None if unset"
    );

    assert_eq!(
        FooBuilder::default().bar(1).bar(2).bar(3).build.unwrap(),
        Foo {[]
            bars: Some(vec![1, 2, 3])
        },
        "Some(Vec<i32>) when set"
    );
}
```

Apologies if I'm missing something to make this work. Otherwise, I think this would be a useful feature to support if possible.

--- a/derive_builder/CHANGELOG.md
+++ b/derive_builder/CHANGELOG.md
@@ -2,6 +2,9 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased]
+- Compatibility between `setter(strip_option`) and `setter(each = "name")` #213
+
 ## [0.10.2] - 2021-04-21
 - Don't reference `derive_builder_core` from `derive_builder` #206
 

--- a/derive_builder/tests/setter_extend.rs
+++ b/derive_builder/tests/setter_extend.rs
@@ -13,6 +13,10 @@ struct Lorem {
     bars: Vec<String>,
     #[builder(setter(each = "baz"))]
     bazes: HashMap<String, i32>,
+    #[builder(setter(strip_option, each = "qux"))]
+    quxes: Option<Vec<String>>,
+    #[builder(setter(strip_option, each = "quux"))]
+    quuxes: Option<HashMap<String, i32>>,
 }
 
 #[derive(Debug, PartialEq, Default, Builder, Clone)]
@@ -24,6 +28,10 @@ struct Ipsum {
     bars: Vec<String>,
     #[builder(setter(each = "baz"))]
     bazes: HashMap<String, i32>,
+    #[builder(setter(strip_option, each = "qux"))]
+    quxes: Option<Vec<String>>,
+    #[builder(setter(strip_option, each = "quux"))]
+    quuxes: Option<HashMap<String, i32>>,
 }
 
 #[test]
@@ -38,6 +46,10 @@ fn extend_field() {
         .baz(("bazz".into(), 2))
         .baz(("bazzz".into(), 3))
         .foo_append("foo")
+        .qux("qux".into())
+        .qux("qux qux".into())
+        .quux(("quux".into(), 1))
+        .quux(("quuxx".into(), 2))
         .build()
         .unwrap();
 
@@ -49,6 +61,12 @@ fn extend_field() {
             bazes: vec![("baz".into(), 1), ("bazz".into(), 2), ("bazzz".into(), 3)]
                 .into_iter()
                 .collect(),
+            quxes: Some(vec!["qux".into(), "qux qux".into()]),
+            quuxes: Some(
+                vec![("quux".into(), 1), ("quuxx".into(), 2)]
+                    .into_iter()
+                    .collect()
+            ),
         }
     );
 }
@@ -65,6 +83,10 @@ fn extend_field_mutable() {
         .baz(("bazz".into(), 2))
         .baz(("bazzz".into(), 3))
         .foo_append("foo")
+        .qux("qux".into())
+        .qux("qux qux".into())
+        .quux(("quux".into(), 1))
+        .quux(("quuxx".into(), 2))
         .build()
         .unwrap();
 
@@ -76,6 +98,12 @@ fn extend_field_mutable() {
             bazes: vec![("baz".into(), 1), ("bazz".into(), 2), ("bazzz".into(), 3)]
                 .into_iter()
                 .collect(),
+            quxes: Some(vec!["qux".into(), "qux qux".into()]),
+            quuxes: Some(
+                vec![("quux".into(), 1), ("quuxx".into(), 2)]
+                    .into_iter()
+                    .collect()
+            ),
         }
     );
 }

--- a/derive_builder_core/src/setter.rs
+++ b/derive_builder_core/src/setter.rs
@@ -159,6 +159,20 @@ impl<'a> ToTokens for Setter<'a> {
             }
 
             if let Some(ref ident_each) = self.each {
+                // Access the collection to extend, initialising with default value if necessary.
+                let field_inner = if stripped_option {
+                    // Outer (builder) Option -> Inner (field) Option -> collection.
+                    quote!(get_or_insert_with(|| Some(
+                        ::derive_builder::export::core::default::Default::default()
+                    ))
+                    .get_or_insert_with(::derive_builder::export::core::default::Default::default))
+                } else {
+                    // Outer (builder) Option -> collection.
+                    quote!(get_or_insert_with(
+                        ::derive_builder::export::core::default::Default::default
+                    ))
+                };
+
                 tokens.append_all(quote!(
                     #(#attrs)*
                     #[allow(unused_mut)]
@@ -168,9 +182,7 @@ impl<'a> ToTokens for Setter<'a> {
                     {
                         #deprecation_notes
                         let mut new = #self_into_return_ty;
-                        new.#field_ident
-                            .get_or_insert_with(::derive_builder::export::core::default::Default::default)
-                            .extend(::derive_builder::export::core::option::Option::Some(item));
+                        new.#field_ident.#field_inner.extend(::derive_builder::export::core::option::Option::Some(item));
                         new
                     }
                 ));

